addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-cidr-service.js

```diff
@@ -0,0 +1,221 @@
+        // This would require storing cidr field in DB
+        // Description: `Updated via SWB`,
+      })),
+      // TODO: Ipv6 calls have to be done separately and cannot habe
+      // Ipv6Ranges: _.map(ipv6List, cidr => ({
+      //   CidrIp: cidr,
+      //   Description: `Updated via SWB`,
+      // })),
+      ToPort: updateRule.toPort,
+    };
+    return IpPermission;
+  }
+
+  // This method is smart enough to determine which IPs to revoke/grant access
+  // by peeking into the ingress rules of the workspace's security group
+  async manageIngressRules(
+    requestContext,
+    { existingEnvironment, securityGroupId, currentIngressRules, updateRequest } = {},
+  ) {
+    const addIpPermissions = [];
+    const revokeIpPermissions = [];
+    const newCidrList = [];
+
+    _.forEach(currentIngressRules, existingRule => {
+      const matchingRule = _.find(updateRequest, {
+        fromPort: existingRule.fromPort,
+        toPort: existingRule.toPort,
+        protocol: existingRule.protocol,
+      });
+      if (matchingRule) {
+        const addCidrs = _.difference(matchingRule.cidrBlocks, existingRule.cidrBlocks);
+        const revokeCidrs = _.difference(existingRule.cidrBlocks, matchingRule.cidrBlocks);
+
+        if (!_.isEmpty(addCidrs)) addIpPermissions.push(this.getIpPermission(matchingRule, addCidrs));
+        if (!_.isEmpty(revokeCidrs)) revokeIpPermissions.push(this.getIpPermission(matchingRule, revokeCidrs));
+        newCidrList.push(matchingRule);
+      }
+    });
+
+    // Get params ready
+    const revokeParams = { GroupId: securityGroupId, IpPermissions: revokeIpPermissions };
+    const authorizeParams = { GroupId: securityGroupId, IpPermissions: addIpPermissions };
+
+    // Start making security group updates
+    const ec2Client = await this.getEc2Client(requestContext, existingEnvironment.id);
+    if (!_.isEmpty(revokeIpPermissions)) await this.revokeSecurityGroupIngress(ec2Client, revokeParams);
+    if (!_.isEmpty(addIpPermissions)) await this.authorizeSecurityGroupIngress(ec2Client, authorizeParams);
+
+    return newCidrList;
+  }
+
+  async revokeSecurityGroupIngress(ec2Client, revokeParams) {
+    try {
+      await ec2Client.revokeSecurityGroupIngress(revokeParams).promise();
+    } catch (err) {
+      // Skip errors that pertain to rule not found
+      if (err.code !== 'EntityNotFound')
+        throw this.boom.internalError(`There was a problem revoking access in security group. Error: ${err}`, true);
+    }
+  }
+
+  async authorizeSecurityGroupIngress(ec2Client, authorizeParams) {
+    try {
+      await ec2Client.authorizeSecurityGroupIngress(authorizeParams).promise();
+    } catch (err) {
+      // Skip errors that pertain to rule already exists
+      if (err.code !== 'AlreadyExists')
+        throw this.boom.internalError(`There was a problem granting access in security group. Error: ${err}`, true);
```

Comment:
Do these errors get logged in Cloudwatch?

Reply:
Yes